scanner/src/main/kotlin/scanners/fossid/FossIdNamingProvider.kt

```diff
@@ -47,26 +49,33 @@ class FossIdNamingProvider(
     }
 
     fun createProjectCode(projectName: String): String = namingProjectPattern?.let {
-        replaceNamingConventionVariables(namingProjectPattern, projectName, namingConventionVariables)
+        val builtins = mapOf(
+            "#projectName" to projectName
+        )
+        replaceNamingConventionVariables(namingProjectPattern, builtins, namingConventionVariables)
     } ?: projectName
 
-    fun createScanCode(projectName: String): String {
-        val pattern = namingScanPattern ?: "#projectName_#currentTimestamp"
-        return replaceNamingConventionVariables(pattern, projectName, namingConventionVariables)
+    fun createScanCode(projectName: String, deltaTag: FossId.DeltaTag? = null): String {
+        val defaultPattern = "#projectName_#currentTimestamp${if (deltaTag == null) "" else "_#deltaTag"}"
+        val pattern = namingScanPattern ?: defaultPattern
+        val builtins = mutableMapOf("#projectName" to projectName)
+        deltaTag?.let {
+            builtins += "#deltaTag" to deltaTag.name.lowercase()
+        }
+        return replaceNamingConventionVariables(pattern, builtins, namingConventionVariables)
```

Comment:
How about checking for `deltaTag != null` only once (and inlining `pattern`), like

        var defaultPattern = "#projectName_#currentTimestamp"
        val builtins = mutableMapOf("#projectName" to projectName)

        if (deltaTag != null) {
            defaultPattern += "_#deltaTag"
            builtins += "#deltaTag" to deltaTag.name.lowercase()
        }

        return replaceNamingConventionVariables(namingScanPattern ?: defaultPattern, builtins, namingConventionVariables)